### ISSUE TITLE
[P0] Mesh-level train/val/test split at DataLoader layer

### ISSUE THREAD

**Issue by AcarBasaran** [ahmetnecirvandogan/sketch-generator#43]
**Priority:** P0 — required before any paper-claimable result
**Scope (narrowed):** mesh-level train/val/test split at the DataLoader layer
**Follow-up:** real Variant A validation + loss-weight ablation now tracked separately

## Why

Without a held-out split, every metric we report on training data is meaningless for paper claims — same garment under different camera/lighting will appear in both train and any "val" set that's just a random sample partition, leaking everything.

The right unit of split is the **mesh**, not the sample. A held-out mesh has *zero* of its samples seen during training, ensuring the network is evaluated on garments it has never seen.

## What to do (sub-task in this issue)

- [ ] Add `split: Literal["train", "val", "test", "all"]` argument to `ClothDataset.__init__` and `make_dataloader` in `pbr_model/dataset.py`.
- [ ] Use deterministic mesh-id hashing (md5 of `mesh_file + seed`) to assign each mesh to a split. Same seed → same partition across runs. Default seed = 42.
- [ ] Default fractions: `val_fraction=0.1`, `test_fraction=0.1`, train ≈ 80 %.
- [ ] Add a `--verify-split` CLI option that builds all three datasets and asserts their mesh sets are disjoint.

## What is NOT in this issue

The remaining sub-tasks from the original #43 scope require real generated data and are now tracked separately:

- Variant A full real-data smoke training (100 steps, all three loss components decreasing without NaN) — needs Stage 1 run on VALAR first
- Loss-weight `λ₁`, `λ₂` ablation — needs first end-to-end training to converge first

These will be filed as a follow-up issue once the loader split lands.

## Files to change

- `pbr_model/dataset.py`

## How to test

```bash
python -m pbr_model.dataset --variant b --split train --verify-split
```

Expected output: prints sample counts for train/val/test and confirms mesh sets are disjoint.

## Estimate

~half a day. ~50 lines in `pbr_model/dataset.py`.
